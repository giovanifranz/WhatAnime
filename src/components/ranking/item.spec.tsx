import { render, screen } from '@/test/test-utils'

import ListItem from './item'

describe('Teste UnitÃ¡rio - ListItem', () => {
  it('Deve renderizar o ListItem corretamente', async () => {
    const mock = {
      index: 3,
      id: 21,
      title: 'Naruto',
    }

    const { asFragment } = render(<ListItem {...mock} />)

    expect(asFragment()).toMatchSnapshot()

    const linkElement = screen.getByText('4. Naruto')

    expect(screen.getByTestId('list-item')).toBeVisible()
    expect(linkElement).toHaveAttribute('href', '/anime/21')
  })
})
