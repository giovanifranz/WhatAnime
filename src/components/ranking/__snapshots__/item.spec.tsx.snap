// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Teste UnitÃ¡rio - ListItem > Deve renderizar o ListItem corretamente 1`] = `
<DocumentFragment>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('dark')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'dark'}catch(e){}}()
  </script>
  <li
    class="w-44 truncate lg:w-56"
    data-testid="list-item"
  >
    <a
      class="transition-all hover:underline hover:opacity-80"
      href="/anime/21"
    >
      4. Naruto
    </a>
  </li>
</DocumentFragment>
`;
